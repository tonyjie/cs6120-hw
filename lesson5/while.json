{
  "functions": [
    {
      "args": [
        {
          "name": "a",
          "type": "int"
        }
      ],
      "instrs": [
        {
          "label": "while.cond"
        },
        {
          "dest": "zero",
          "op": "const",
          "type": "int",
          "value": 0
        },
        {
          "args": [
            "a",
            "zero"
          ],
          "dest": "is_term",
          "op": "eq",
          "type": "bool"
        },
        {
          "args": [
            "is_term"
          ],
          "labels": [
            "while.finish",
            "while.body"
          ],
          "op": "br"
        },
        {
          "label": "while.body"
        },
        {
          "dest": "one",
          "op": "const",
          "type": "int",
          "value": 1
        },
        {
          "args": [
            "a",
            "one"
          ],
          "dest": "a",
          "op": "sub",
          "type": "int"
        },
        {
          "labels": [
            "while.cond"
          ],
          "op": "jmp"
        },
        {
          "label": "while.finish"
        },
        {
          "args": [
            "a"
          ],
          "op": "print"
        }
      ],
      "name": "main"
    }
  ]
}
