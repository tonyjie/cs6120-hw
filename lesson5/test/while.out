Dom: {'entry.insert': {'entry.insert'}, 'while.cond': {'while.cond', 'entry.insert'}, 'while.body': {'while.cond', 'entry.insert', 'while.body'}, 'while.finish': {'while.cond', 'entry.insert', 'while.finish'}}

Dom Tree: {'entry.insert': ['while.cond'], 'while.cond': ['while.body', 'while.finish'], 'while.body': [], 'while.finish': []}

Dom Frontier: {'entry.insert': [], 'while.cond': ['while.cond'], 'while.body': ['while.cond'], 'while.finish': []}

